openfpga/src/utils/report_reference.cpp

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************************
+ * Output internal structure of module graph to XML format
+ ***************************************************************************************/
+/* Headers from system goes first */
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <string>
+
+/* Headers from vtrutil library */
+#include "vtr_assert.h"
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+/* Headers from openfpgautil library */
+#include "command_exit_codes.h"
+#include "openfpga_digest.h"
+#include "report_reference.h"
+
+/* begin namespace openfpga */
+namespace openfpga {
+
+/********************************************************************
+ * Top-level function
+ *******************************************************************/
+int report_reference(const char* fname, const std::string& module_name,
+                     const ModuleManager& module_manager,
+                     const bool& include_time_stamp, const bool& verbose) {
+  vtr::ScopedStartFinishTimer timer("Report reference");
+
+  std::fstream fp;
+  fp.open(std::string(fname), std::fstream::out | std::fstream::trunc);
+  openfpga::check_file_stream(fname, fp);
+
+  if (include_time_stamp) {
+    auto end = std::chrono::system_clock::now();
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end);
+    fp << "Date: " << std::ctime(&end_time) << std::endl;
+  }
+
+  ModuleId parent_module = module_manager.find_module(module_name);
+  if (ModuleId::INVALID() == parent_module) {
```

Comment:
Use existing API to validate the id: 

https://github.com/lnis-uofu/OpenFPGA/blob/feefc04d14fd323f864431d1e065f6fd09908502/openfpga/src/fabric/module_manager.h#L535